### PR TITLE
pass device_str for async_compile.triton function

### LINKED ISSUE #120203: pass device_str for async_compile.triton function
### 🚀 The feature, motivation and pitch

When [codegening](https://github.com/pytorch/pytorch/blob/main/torch/_inductor/codecache.py#L2570), the real device was not passed to the triton function for execution. Here, we add a string parameter for the device to ensure the authenticity of the coding

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by wangjiangben-hw**
Fixes #120203

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler @amjames @jansel

### CODE REVIEW

**Review by jansel — CHANGES_REQUESTED** [review]
What happens if a graph contains a mix of devices?  For example CPU +CUDA?  Two GPUs?

**Comment by wangjiangben-hw** [comment]
> What happens if a graph contains a mix of devices? For example CPU +CUDA? Two GPUs?

Currently, Triton is also developing multiple backends. We can see that XPU and CPU are preparing for related matters, https://github.com/openai/triton/pull/2551. The modification here is only to align the input parameters with "device_type" in triton_meta. This ensures that the triton function correctly transfers parameters during compilation.

File: torch/_inductor/codegen/triton.py [hunk 1]
```diff
@@ -3503,7 +3503,7 @@ def define_kernel(self, src_code, node_schedule):
             compile_wrapper = IndentedBuffer()
             compile_wrapper.writeline(f"async_compile.triton({subs_name!r}, '''")
             compile_wrapper.splice(src_code, strip=True)
-            compile_wrapper.writeline("''')")
+            compile_wrapper.writeline(f"''', device_str='{V.graph.scheduler.current_device.type}')")
```
[the remark this answers is not in the document]
Comment (wangjiangben-hw) [reply torch/_inductor/codegen/triton.py:3506]: For GPU, yes, it is always cuda. 
If triton supports other devices in the future, the parameter may be incorrect.
I modified it like this, for the cuda device, there will be no change, just to supplement the parameters to be passed down.

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**